ambry-api/src/main/java/com/github/ambry/config/NetworkConfig.java

```diff
@@ -119,6 +122,35 @@ public class NetworkConfig {
   @Default("false")
   public final boolean selectorUseDirectBuffers;
 
+  /**
+   * The type of queue to use to hold requests waiting to be processed.
+   * TODO: Do tests to tune controlled delay queue parameters before switching the request queue type to {@link RequestQueueType#ADAPTIVE_LIFO_CO_DEL}
+   */
+  @Config("request.queue.type")
+  @Default("BASIC_FIFO")
+  public final RequestQueueType requestQueueType;
+
+  /**
+   * The maximum time in milliseconds a request is allowed to spend in the queue waiting to be processed.
+   */
+  @Config("request.queue.timeout.ms")
+  @Default("2000")
```

Comment:
how do u decide these defaults ?

Reply:
Discussed with @justinlin-linkedin offline on coming up with this defaults. It is related to our request timeout at frontends and request processing time at server. Since our frontends mostly have request time out of 4 seconds and if we assume request processing time around 1.5 second (based on P95 values in our graphs) and give another 0.5 seconds to network trip, we can probably allow 2 seconds for the requests in the queue before declaring them stale.